src/config.rs

```diff
@@ -66,12 +66,17 @@ impl Config {
             .map(|td| td.url.as_str())
             .unwrap_or_default();
 
-        let url = match (default_url.as_str(), token_url) {
-            (_, "") => default_url,
-            _ if default_url == token_url => default_url,
-            (DEFAULT_URL | "", _) => String::from(token_url),
+        let region_url = token_embedded_data
+            .as_ref()
+            .map(|td| td.region_url.as_str())
+            .unwrap_or_default();
+
+        let url = match (default_url.as_str(), token_url, region_url) {
+            (_, "", "") => default_url,
+            _ if default_url == token_url || default_url == region_url => default_url,
+            (DEFAULT_URL | "", _, _) => String::from(token_url),
             _ => bail!(
-                "Two different url values supplied: `{token_url}` (from token), `{default_url}`."
+                "URL must match one of the provided URLs on the authentication token: `{token_url}` or `{region_url}`, received: `{default_url}`."
```

Comment:
This could probably use some better wording, but it feels appropriate to include the region URL here as an option for the user to select it.

Reply:
Also this is a little quirky. If the region_url and token_url match, you'll essentially get a message with the same expected URL twice. I can make this exhaustive at the expense of adding multiple distinct error message cases.

Reply:
Furthermore, this change only affects a URL supplied in a configuration file or environment variable. Unfortunately, the logic to set the URL from a command line argument is located in the function `set_base_url`.

Reply:
Yeah, I think we should add the distinct error message cases; might be confusing if we list the same URL twice in the same error message